sympy/physics/control/lti.py

```diff
@@ -0,0 +1,181 @@
+from sympy import Basic, Mul, degree, Symbol, expand, cancel, Expr
+from sympy.core.numbers import Integer, Float
+from sympy.core.sympify import sympify, _sympify
+
+__all__ = ['TransferFunction',]
+
+
+class TransferFunction(Basic):
+
+    def __new__(cls, num, den, var):
+        num, den = _sympify(num), _sympify(den)
+
+        if not isinstance(var, Symbol):
+            raise TypeError("Variable input must be a Symbol.")
+        if den == 0:
+            raise ValueError("TransferFunction can't have a zero denominator.")
+
+        if (isinstance(num, Expr) and num.has(Symbol)) or isinstance(num, Integer):
+            if (isinstance(num, Expr) and num.has(Symbol)) or isinstance(num, Integer):
+                obj = super(TransferFunction, cls).__new__(cls, num, den, var)
+                obj._num = num
+                obj._den = den
+                obj._var = var
+                return obj
+        else:
+            raise ValueError("Unsupported type for numerator or denominator of TransferFunction.")
+
+    @property
+    def num(self):
+        return self._num
+
+    @property
+    def den(self):
+        return self._den
+
+    @property
+    def var(self):
+        return self._var
+
+    def simplify(self):
+        tf = cancel(Mul(self.num, 1/self.den)).as_numer_denom()
+        num_, den_ = tf[0], tf[1]
+        return TransferFunction(num_, den_, self.var)
+
+    def expand(self):
+        return TransferFunction(expand(self.num), expand(self.den), self.var)
+
+    def __add__(self, other):
+        other = _sympify(other)
+        if isinstance(other, (Integer, Float)):
+            if isinstance(other, Float):
+                other = Float(other, 3)
```

Comment:
Any reason for limiting the precision here?

Reply:
Creating Floats from Python `float` type gives a minimum precision of 15 digits

Reply:
If you have intended to implement feature to add with constants, I don’t think that it should be limited to Integer or Float, but to support any constants via other.is_number.

And I don’t think that we should modify precisions here, but dispatch everything to how sympy addition and multiplication handles floats.

Reply:
> I don’t think that it should be limited to Integer or Float, but to support any constants via other.is_number.

I think we have to use `isinstance(other, (Integer, Float))` because `other.is_number` might allow other objects. For isinstance:
``` python
>>> q = Quaternion(1, 2, 3, 4)
>>> q.is_number
True
```

Reply:
What about a symbolic constant?